planner/core/logical_plan_builder.go

```diff
@@ -4166,6 +4177,20 @@ func (b *PlanBuilder) tryBuildCTE(ctx context.Context, tn *ast.TableName, asName
 			lp := LogicalCTE{cteAsName: tn.Name, cte: cte.cteClass, seedStat: cte.seedStat, isOuterMostCTE: !b.buildingCTE}.Init(b.ctx, b.getSelectOffset())
 			prevSchema := cte.seedLP.Schema().Clone()
 			lp.SetSchema(getResultCTESchema(cte.seedLP.Schema(), b.ctx.GetSessionVars()))
+
+			if cte.isInline {
+				lp.MergeHints.preferMerge = cte.isInline
```

Comment:
Why set it here?

Reply:
When a query about CTE with `merge` hint, We are here to decide whether to use inline on this query as well as let it go to 'buildmerge'. And then the query will not go to buildCTE.